### ISSUE TITLE
Relative knobs

### ISSUE THREAD

**Issue by git-moss** [electraone/firmware#15]
Allow knobs to send relative values. Since the encoding of relative values is already available in the editor, a switch to choose between absolute/relative should be sufficient.
If the knobs can detect speed it would be nice if this could be encoded too.

**Comment by electraone** [comment]
A a simple version of the relative control have been added. I am wondering if this should be supported with additional functionality such as setting the "absolute" value by host application, etc.

**Comment by git-moss** [comment]
The new relative knob works great for values which do not make sense with a fader like tempo or play position!
Nevertheless, it would be great to have the option on the fader element as well. In that case the value (and by this the fader) and value label should only be changed via the sysex commands.

**Comment by electraone** [comment]
Yes, I was expecting this coming in but I preferred to hear from you. I propose the following: I would add an extra call to override the display value. It means there would be a sysex call to set the value (within the boundaries of the display min and max).

**Comment by git-moss** [comment]
Sorry, don't get your comment. The display value can aleady be set since 3.1?!

**Comment by electraone** [comment]
yes and no. The recently added calls set the text to be displayed instead of the value. ie. it does not carry the numeric information. The numeric value is needed to draw the component (eg fader) correctly. If you sent the value text (recent API calls) the actual fader track will not be updated.

So in case of, say fader, when set to relative mode, it would emit the CC inc/decrement MIDI messages to the host application. And the host application would in return set the numeric and possibly the value text.

Does that make sense to you?

**Comment by git-moss** [comment]
Yes, exactly. But I can also already set the value with MIDI CC. So, I think we only need a CC relative option on the fader control:
- Element keeps the value state but does not send it
- Element sends instead the -x/+x values
- Value for fader drawing is sent with MIDI CC
- Value label with newly introduced sysex command

**Comment by electraone** [comment]
That makes sense. I will include it in next release.
